### ISSUE TITLE
Question: Allowing at most 2 jobs of the same type

### ISSUE THREAD

**Issue by igorkulman** [lucas34/SwiftQueue#210]
I found `func singleInstance(forId: String, override: Bool = false) -> Self`  to allow only 1 job of a type but my use case is a bit different.

I do a sync with the backend and during the app lifetime there are requests for that sync coming in. The idea is that when a sync is already running, it should be not canceled and another sync should be scheduled. But when one sync is already running and another one is already scheduled, there is no point in scheduling a third sync. 

Basically I need the queue to always have at most 1 running and 1 scheduled operation. Is this possible?

**Comment by lucas34** [comment]
I understand your requirement, Right now this use case is not supported however, It's a bit tricky but should be possible to implement. I'll have a look when I got time

**Comment by lucas34** [comment]
Included in 3.1.0
